src/Microsoft.AspNetCore.Http.Abstractions/Routing/RouteValueDictionary.cs

```diff
@@ -146,7 +154,10 @@ public RouteValueDictionary(object values)
                 var storage = new PropertyStorage(values);
                 _propertyStorage = storage;
                 _count = storage.Properties.Length;
```

Comment:
Doesn’t this case leave the `_arrayStorage` uninitialized now? All other cases have an explicit initialization but this case does not.

Reply:
It does indeed leave array storage uninitialized, but that's not an issue as either:
* property storage is initialized and used
-or-
* array storage is initialiazed and used

When initialized, we always prefer property storage over array storage. Array storage can remain uninitialized when property storage is initialized.

When property storage is initialized, but we want to manipulate the values, then the property storage is transformed to array storage. This is done using [`RouteValueDictionary.EnsureCapacitySlow(int capacity)`](https://github.com/aspnet/HttpAbstractions/blob/c9be018c7e2cce41157b37a17da7189d772635bc/src/Microsoft.AspNetCore.Http.Abstractions/Routing/RouteValueDictionary.cs#L544).

Reply:
Ah cool, I missed that every use of the array storage would ensure the capacity before which would cause an initialization of the storage.

Since the checks always check whether the property storage is set (instead of checking when the array storage is _not_ set), does it really make such a difference to not initialize the array storage to an empty array by default? All those `_arrayStorage = Array.Empty…` lines really seem verbose when they are always executed except in the one case where an explicit array is being set, or when the property storage is being set which does not need the array storage to be _not_ set though.

Reply:
I just thought it would be clearer when either `_propertyStorage` or `_arrayStorage` would be initialized, and not both.